src/nunit.analyzers.tests/DisposeFieldsInTearDown/DisposeFieldsInTearDownAnalyzerTests.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections.Generic;
+using Gu.Roslyn.Asserts;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.DisposeFieldsInTearDown;
+using NUnit.Framework;
+
+namespace NUnit.Analyzers.Tests.DisposeFieldsInTearDown
+{
+    public class DisposeFieldsInTearDownAnalyzerTests
+    {
+        private const string DummyDisposable = @"
+        private sealed class DummyDisposable : IDisposable
+        {
+            public void Dispose() {}
+        }";
+
+        // CA2000 allows transfer of ownership using ICollection<IDisposable>.Add
```

Comment:
I didn't know about this, but some googling gave the same answer - e.g. https://stackoverflow.com/a/3966041 - but I could not find an official documentation of this

Reply:
The only thing official is the source code: [`IsCollectionAddMethod`](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs#L465-L499). From that, the method should be either `Add` or `AddOrUpdate` even though the latter is not part of `ICollection`. Go figure.

We use the `Disposer` pattern a lot to ensure items that need disposable are added where they are created.